src/utils.js

```diff
@@ -220,6 +220,14 @@ function toSentence(
   )
 }
 
+function isEqualWithArraysAsSets(arr1, arr2) {
+  if (Array.isArray(arr1) && Array.isArray(arr2)) {
+    return [...new Set(arr1)].every(v => new Set(arr2).has(v))
+  } else {
+    return arr1 === arr2
+  }
+}
```

Comment:
I'm now having second thoughts about discarding the use of lodash's `isEqual` entirely. For a few reasons:

- It is a battle-tested implementation that covers edge cases that we might not be covering here (see below for one of these cases).
- It is not featured in the website https://youmightnotneed.com/lodash/, meaning that they do not have any recommended in-house implementation that replaces the lodash function.

## Edge case not covered here.

I think that the function in lodash is probably recursive. In the `else` part it probably calls itself up:

```suggestion
function isEqualWithArraysAsSets(arr1, arr2) {
  if (Array.isArray(arr1) && Array.isArray(arr2)) {
    return [...new Set(arr1)].every(v => new Set(arr2).has(v))
  } else {
    return isEqualWith(arr1, arr2)
  }
}
```

Also, probably the lodash implementation avoids a potential infinite recursion problem if an object has references to part of itself, like this:

```js
const obj = { list: [obj] }
```

But probably we do not need to cover that edge case here. So maybe doing the change suggested above is enough, and we can indeed ditch lodash entirely.

What do you think?